tests/test_branch.py

```diff
@@ -0,0 +1,168 @@
+#!/usr/bin/env python3
+from contextlib import contextmanager
+from datetime import datetime, timedelta
+import os
+from pathlib import Path
+from tempfile import TemporaryDirectory
+import subprocess
+import unittest.mock
+from typing import Any, BinaryIO, Iterator
+from nix_update.version import fetch_latest_version
+from nix_update.version.version import VersionPreference
+import conftest
+
+TEST_ROOT = Path(__file__).parent.resolve()
+
+
+def create_dummy_commit(tmp: str, date: datetime) -> None:
+    datestr = date.strftime("%FT%TZ")
+    with open(Path(tmp) / "date.txt", "w", encoding="utf-8") as fp:
+        fp.write(datestr)
+        fp.write("\n")
+    subprocess.run(["git", "-C", tmp, "add", "date.txt"], check=True)
+    env = os.environ.copy()
+    env["GIT_AUTHOR_DATE"] = datestr
+    env["GIT_COMMITTER_DATE"] = datestr
+    subprocess.run(
+        [
+            "git",
+            "-C",
+            tmp,
+            "-c",
+            "user.name=dummy",
+            "-c",
+            "user.email=dummy@localhost",
```

Comment:
```suggestion
            "user.email=<dumm> dummy@localhost",
```

Reply:
I'm not sure what you mean.

Reply:
Oh, I mixed this string up with the combined name and mail string.

Reply:
Ah ok, that shouldn't be necessary, git takes care of that and builds the SMTP-like author from user.name and user.email.